### PR TITLE
Manage XCM through maintenance mode filters

### PR DESCRIPTION
**Pull request by Agusrodri**
This PR adds the possibility of managing XCM functionalities through maintenance mode filters.

The following changes were applied to frontier and simple templates, and also to the dancebox runtime:

* `MaintenanceFilter` will forbid any call to `polkadotXcm` pallet.

* `xcm-support` feature was added to `pallet-maintenance-mode`, alongside with the custom implementations of `XcmExecutionManager`, `NormalDmpHandler` and `MaintenanceDmpHandler` types.

### CODE REVIEW

File: container-chains/templates/simple/runtime/src/lib.rs [hunk 1]
```diff
@@ -432,8 +445,43 @@ impl Contains<RuntimeCall> for MaintenanceFilter {
 /// This can change in the future
 pub struct NormalFilter;
 impl Contains<RuntimeCall> for NormalFilter {
-    fn contains(_c: &RuntimeCall) -> bool {
-        true
+    fn contains(c: &RuntimeCall) -> bool {
+        match c {
+            // Only allow force_default_xcm_version
+            RuntimeCall::PolkadotXcm(method) => match method {
```
Comment (girazoki) [inline container-chains/templates/simple/runtime/src/lib.rs:451]: This is actually not something we need, as we dont want to prohibit calls to polkadotXcm

File: test/suites/dev-frontier-template/test-maintenance/test-maintenance-xcm-normal-filter.ts [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+                const dest = {
+                    V3: destMultilocation,
+                };
+
+                const message = {
+                    V3: [{ ClearOrigin: null }],
+                };
+
+                const sendTx = context.polkadotJs().tx.polkadotXcm.send(dest, message);
```
Comment (girazoki) [inline test/suites/dev-frontier-template/test-maintenance/test-maintenance-xcm-normal-filter.ts:67]: These are pretty much the same test, we should test it in common

File: container-chains/templates/frontier/runtime/src/lib.rs [hunk 3]
```diff
@@ -520,10 +522,20 @@ impl pallet_utility::Config for Runtime {
     type WeightInfo = pallet_utility::weights::SubstrateWeight<Runtime>;
 }
 
+pub struct XcmExecutionManager;
```
Comment (girazoki) [inline container-chains/templates/frontier/runtime/src/lib.rs:525]: I am starting to think we should move this into xcm-pritmitives

**Comment by Agusrodri** [comment]
TODO: add tests for DMP and general XCM messages in maintenance mode.

**Comment by girazoki** [comment]
@Agusrodri can you merge master in? Also we can merge as is but I will create an issue to move these into their proper primitives crates in moonkit

**Review by girazoki — APPROVED** [review]

**Pull request merged**